src/backend/routers/agent.py

```diff
@@ -42,7 +70,20 @@ def create_agent(session: DBSessionDep, agent: CreateAgent, request: Request) ->
 
     request.state.agent = agent_data
     try:
-        return agent_crud.create_agent(session, agent_data)
+        created_agent = agent_crud.create_agent(session, agent_data)
+        if agent.tools_metadata:
```

Comment:
To be on the same page the payload would look like
```
[
    {
        "tool_name": "name1",
        "type": "type1",
        "artifacts": ["id1", "id2"]
    }
]
```

Reply:
yes, i deviated a bit from what we spoke about https://github.com/cohere-ai/cohere-toolkit/pull/309#issuecomment-2195844024

i think its a lot easier to read and parse like this. it also reflects what its actually stored in the DB exactly and works well with the schema in the codebase.